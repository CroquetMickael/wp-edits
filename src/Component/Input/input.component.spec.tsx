import React from "react";
import { render, RenderResult, cleanup } from "react-testing-library";
import InputComponent from "./input.component";

interface InputProps
  extends Readonly<{
    label: string;
    name: string;
    value: string;
    handler: (event: any) => void;
  }> {}

describe("Input Component", () => {
  let component: RenderResult;
  let props: InputProps;
  beforeEach(() => {
    let value = "Mickael";
    let label = "Name";
    let name = "firstname";
    let handler = () => jest.fn();
    props = {
      value,
      handler,
      label,
      name
    };
    component = render(<InputComponent {...props} />);
  });
  afterEach(cleanup);

  it("Should render properly Input Component", () => {
    const { container } = component;
    expect(container).toMatchSnapshot();
  });
});
