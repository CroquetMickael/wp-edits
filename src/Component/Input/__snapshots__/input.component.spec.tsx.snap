// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Should render properly Input Component 1`] = `
<div>
  <div
    class=""
  >
    <input
      class="appearance-none border-b-2 border-b border-grey bg-transparent 
 text-grey-darker m-2 py-2 leading-tight 
 focus:outline-none focus:border-blue w-full"
      name="firstname"
      placeholder="Name"
      required=""
      type="text"
      value="Mickael"
    />
  </div>
</div>
`;
